kms/kms.rb

```diff
@@ -0,0 +1,392 @@
+# Copyright 2017 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in write, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Note: Code samples in this file set constants, and
+#       constants cannot be set inside method definitions in Ruby,
+#       code snippets in this sample are wrapped in global lambdas.
+
+$create_keyring = -> (project_id:, key_ring_id:, location:) do
+  # [START kms_create_keyring]
+  # project_id  = "Your Google Cloud project ID"
+  # key_ring_id = "The id of the new KeyRing"
+  # location    = "The location of the new KeyRing"
+
+  require "google/apis/cloudkms_v1beta1"
+
+  # Instantiate the client, authenticate with specified scope
+  Cloudkms = Google::Apis::CloudkmsV1beta1
+  kms_client = Cloudkms::CloudKMSService.new
+  kms_client.authorization = Google::Auth.get_application_default(
+    "https://www.googleapis.com/auth/cloud-platform"
+  )
+
+  # The resource name of the location associated with the KeyRing
+  resource = "projects/#{project_id}/locations/#{location}"
+
+  # Create a KeyRing for your project
+  key_ring = kms_client.create_project_location_key_ring(resource,
```

Comment:
This syntax of parameter wrapping is a bit non-standard, and I think it would be the only place in ruby-docs-samples where we use it:

```ruby
call_method(param1,
  param2, param3)
```

Usually, if your method params go onto multiple lines, I tend to lean towards this syntax for clarity:

```ruby
call_method(
  param1,
  param2,
  param3
)
```

Even though it adds more lines 😕 

Man, this sure is a wordy API!  Hate to ask you to change all of these, but I think it would be easier to comprehend and matches the [few other](https://github.com/GoogleCloudPlatform/ruby-docs-samples/blob/04edeef14e1ddd4abc994bccd46ea23e1d85c35d/language/spec/language_samples_spec.rb#L94-L97) [places](https://github.com/GoogleCloudPlatform/ruby-docs-samples/blob/9acdd156e98d43c333d98b7e5d0c413de17c41a4/datastore/sample.rb#L47-L51) where we use parens like this.

(*We don't have many parenthesis in this repo!* 😄 )